lib/cli-engine.js

```diff
@@ -429,6 +428,24 @@ class CLIEngine {
              */
             this._lintResultCache = new LintResultCache(cacheFile, this.config);
         }
+
+        // setup special filter for fixes
+        if (this.options.fixTypes && this.options.fixTypes.length > 0) {
+
+            // convert to Set for faster lookup
+            const fixTypes = new Set(this.options.fixTypes);
+
+            // save original value of options.fix in case it's a function
+            const originalFix = this.options.fix;
+
+            this.options.fix = lintResult => {
+                const rules = this.getRules();
```

Comment:
Ah thanks, I'll look into that.

Reply:
So I moved `getRules()` out but forgot about how rules can be added later. Hmmm. That's a tricky one.

Reply:
Okay, here's the approach I took: Create a `_rulesCache` property that is populate each time `executeOnText()` or `executeOnFiles()` is called. That way, it's called just once before doing the operation instead of each time a lint result needs to be checked. That should be much more efficient, and deal with new rules being added.